src/Data/Yaml.hs

```diff
@@ -66,6 +68,10 @@ module Data.Yaml
       -- * Classes
     , ToJSON (..)
     , FromJSON (..)
+      -- * Custom encoding
+    , isSpecialString
+    , EncodeOptions(..)
```

Comment:
To avoid backwards incompatible changes in the future, I'd rather not expose the data constructors here. Instead, I'd like to see something like:

```haskell
, EncodeOptions
, defaultEncodeOptions
, setEncodeOptionsStringStyle
```